Firmware/FFBoard/Src/ShifterAnalog.cpp

```diff
@@ -65,58 +76,188 @@ void ShifterAnalog::readButtons(uint32_t* buf){
 			  gear=4;       // 4th gear
 		}
 
-		if(gear == 6 && (HAL_GPIO_ReadPin(rev_port,rev_pin) == GPIO_PIN_SET)){
+		if(gear == 6 && reverseButtonState){
 			gear = 7; // Reverse
 		}
 	}
+}
 
-	if(gear > 0){
-		*buf = 1 << (gear-1);
+void ShifterAnalog::updateReverseState() {
+	if (g27ShifterButtonClient) {
+		reverseButtonState = g27ShifterButtonClient->getReverseButton();
+	} else {
+		reverseButtonState = LocalButtons::readButton(reverseButtonNum);
```

Comment:
This is a little questionable...it may make sense to separate out the concept of mapping the digital inputs to pins/ports from the `LocalButtons` class.

Reply:
Maybe. But i think its fine at the moment because this class maps the digital inputs and the shifter is connected to a button input.